batch/batches/StorageUpdate/KAsyncStorageUpdate.php

```diff
@@ -0,0 +1,443 @@
+		if(self::isPartnerCreatedAsMonitoredFreeTrial($partner))
+		{
+			$monitoredFreeTrial = true;
+			if ($partner->partnerPackage == KAsyncStorageUpdateUtils::PARTNER_PACKAGE_DEVELOPER_TRIAL)
+			{
+				$divisionFactor = $partnerPackage['cycle_bw_for_monitored_trial'];
+			}
+		}
+
+		list($percent, $totalUsageGB) = self::getPercentFromStatistics($partner, $divisionFactor);
+		$systemPartnerConfiguration->usagePercent = $percent;
+
+		KalturaLog::debug('percent ('.$partner->id.') is: '.$percent);
+		$emailLinkHash = 'pid='.$partner->id.'&h='.(self::getEmailLinkHash($partner->id, $partner->secret));
+
+		$mindtouchNotice = ' ';
+		if($partner->type == KAsyncStorageUpdateUtils::DEKIWIKI) // dekiwiki-mindtouch partner
+		{
+			$mindtouchNotice = '<BR><BR>Note: You must be a MindTouch paying customer to upgrade your video service. If you are not a paying MindTouch customer, contact MindTouch: http://www.mindtouch.com/about_mindtouch/contact_mindtouch to get a quote.<BR><BR>';
+		}
+
+		$this->handleUsage($percent, $partner, $systemPartnerConfiguration, $monitoredFreeTrial, $partnerPackage, $mindtouchNotice, $totalUsageGB, $emailLinkHash, $shouldBlockDeletePartner, $blockNotificationGrace, $deleteGrace);
+	}
+
+	public function getRelevantFields($partner)
+	{
+		$shouldBlockDeletePartner = true;
+		$blockNotificationGrace = time() - (KAsyncStorageUpdateUtils::DAY * KAsyncStorageUpdateUtils::BLOCKING_DAYS_GRACE);
+		$deleteGrace = time() -  (KAsyncStorageUpdateUtils::DAY * 30);
+		$partnerPackage = $this->packages[$partner->partnerPackage];
+		$divisionFactor = $partnerPackage['cycle_bw'];
+		return array($shouldBlockDeletePartner, $blockNotificationGrace, $deleteGrace, $partnerPackage, $divisionFactor);
+	}
+
+	public function getPercentFromStatistics($partner, $divisionFactor)
+	{
+		$divisionFactor = ($divisionFactor != 0 ? $divisionFactor : 1);
+		KBatchBase::impersonate($partner->id);
+		$partnerStatistics = self::$kClient->partner->getStatistics();
+		KBatchBase::unimpersonate();
+		$totalUsageGB = $partnerStatistics->usage;
+		$percent = round( ($totalUsageGB / $divisionFactor)*100, 2);
+		return array($percent, $totalUsageGB);
+	}
+
+	public function handleUsage($percent, $partner, $systemPartnerConfiguration, $monitoredFreeTrial, $partnerPackage, $mindtouchNotice, $totalUsageGB, $emailLinkHash, $shouldBlockDeletePartner, $blockNotificationGrace, $deleteGrace)
+	{
+		if ($percent < 80 )
+		{
+			if ($partner->eightyPercentWarning)
+			{
+				KalturaLog::debug('partner '. $partner->id .' was 80%, now not. clearing warnings');
+				$systemPartnerConfiguration->eightyPercentWarning = 0;
+				$systemPartnerConfiguration->usageLimitWarning = 0;
+			}
+			elseif ($partner->usageLimitWarning)
+			{
+				KalturaLog::debug('partner '. $partner->id .' OK');
+				$systemPartnerConfiguration->eightyPercentWarning = 0;
+				$systemPartnerConfiguration->usageLimitWarning = 0;
+			}
+		}
+		elseif ($percent >= 80 && $percent < 100)

```

Comment:
for all the values, use const 
const WATERMARK_HIGH = 100
const WATERMARK_LOW = 80